File: Courzelo-Base/src/main/java/tn/esprit/devflow/courzelo/entity/Class.java
package tn.esprit.devflow.courzelo.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.ArrayList;
import java.util.List;

@Data
@Document
@FieldDefaults(level= AccessLevel.PRIVATE)

public class Class {
    @Id
     String idClass;
     int capacity;
    @Field("level")
     Level level;
    Integer progress;
    @Field("speciality")
    Speciality speciality;
    public void addLesson(Lesson lesson) {
        if (lessons == null) {
            lessons = new ArrayList<>();
        }
        lessons.add(lesson);
    }

    @DBRef
     User user;
//    @JsonIgnore
    @DBRef
     List<Module> modules;
    @DBRef
    List<Lesson> lessons;
}

File: Courzelo_Angular/src/app/models/Class/class.ts
import { Lesson } from "../Lesson/lesson";

export class Class {
    idClass?: string;
    capacity?: number;
    level?: Level;
    progress?: number;
    speciality?: Speciality;
    lessons?: Lesson[];
}



export enum Speciality{
    Informatique='Informatique'
    ,ElectroMecanique='ElectroMecanique'
    ,GenieCivil='GenieCivil'
}
export enum Level{
    LEVEL_1='LEVEL_1'
    ,LEVEL_2='LEVEL_2'
    ,LEVEL_3='LEVEL_3'
    ,LEVEL_4='LEVEL_4'
    ,LEVEL_5='LEVEL_5'
   

}